shared/util/maps.go

```diff
@@ -0,0 +1,168 @@
+package util
```

Comment:
What do you think about moving maps.go to a new package in the go-misc repo?

Reply:
Sounds like a good idea; I can do it in a followup PR.